internal/kibana/prebuilt_rules/models.go

```diff
@@ -0,0 +1,245 @@
+	var resp *kbapi.InstallPrebuiltRulesAndTimelinesResponse
+	var err error
+
+	if spaceID != "default" {
+		resp, err = client.API.InstallPrebuiltRulesAndTimelinesWithResponse(ctx, func(ctx context.Context, req *http.Request) error {
+			req.Header.Set("kbn-space-id", spaceID)
+			return nil
+		})
+	} else {
+		resp, err = client.API.InstallPrebuiltRulesAndTimelinesWithResponse(ctx)
+	}
+
+	if err != nil {
+		return utils.FrameworkDiagFromError(err)
+	}
+
+	if resp.StatusCode() != 200 {
+		return utils.FrameworkDiagFromError(fmt.Errorf("failed to install prebuilt rules: %s", resp.Status()))
+	}
+
+	return nil
+}
+
+func needsRuleUpdate(ctx context.Context, client *kibana_oapi.Client, spaceID string) bool {
+	status, diags := getPrebuiltRulesStatus(ctx, client, spaceID)
+	if diags.HasError() {
+		return true
+	}
+	return status.RulesNotInstalled >= 1 || status.RulesNotUpdated >= 1
+}
+
+func manageRulesByTags(ctx context.Context, client *kibana_oapi.Client, spaceID string, tags []string) diag.Diagnostics {
+	// Reject "all" as it's not supported by Kibana for large rule sets
+	if len(tags) == 1 && tags[0] == "all" {
+		return utils.FrameworkDiagFromError(fmt.Errorf("enabling all rules is not supported due to Kibana API limitations. Please specify specific tags to enable a subset of rules"))
+	}
+
+	if len(tags) == 0 {
+		// If no tags specified, this resource manages no rules - this is valid
+		return nil
+	}
+
+	// Enable rules matching the specified tags
+	diags := performBulkActionByTags(ctx, client, spaceID, "enable", tags)
+	if diags.HasError() {
+		return diags
+	}
+
+	return nil
+}
+
+func manageRulesTagTransition(ctx context.Context, client *kibana_oapi.Client, spaceID string, oldTags, newTags []string) diag.Diagnostics {
+	// Handle tag transitions for declarative behavior
+
+	// Find tags that were removed (need to disable their rules)
+	var removedTags []string
+	for _, oldTag := range oldTags {
+		found := false
+		for _, newTag := range newTags {
+			if oldTag == newTag {
+				found = true
+				break
+			}
+		}
+		if !found {
+			removedTags = append(removedTags, oldTag)
+		}
+	}
+
+	// Find tags that were added (need to enable their rules)
+	var addedTags []string
+	for _, newTag := range newTags {
+		found := false
+		for _, oldTag := range oldTags {
+			if newTag == oldTag {
+				found = true
+				break
+			}
+		}
+		if !found {
+			addedTags = append(addedTags, newTag)
+		}
+	}
+
+	// Disable rules for removed tags
+	if len(removedTags) > 0 {
+		diags := performBulkActionByTags(ctx, client, spaceID, "disable", removedTags)
+		if diags.HasError() {
+			return diags
+		}
+	}
+
+	// Enable rules for added tags
+	if len(addedTags) > 0 {
+		diags := performBulkActionByTags(ctx, client, spaceID, "enable", addedTags)
+		if diags.HasError() {
+			return diags
+		}
+	}
+
+	return nil
+}
+
+func performBulkActionByTags(ctx context.Context, client *kibana_oapi.Client, spaceID, action string, tags []string) diag.Diagnostics {
```

Comment:
I'm certainly unfamiliar with this API, so apologies if I'm missing something obvious. What prevents this resource from operating on user defined rules that also match the given tag set? That is, what restricts the resource to only enabling 'pre-built' rules, and not user defined rules which also match a given tag?